app/Main.hs

```diff
@@ -465,6 +472,55 @@ verifyPackage db paths name = do
   let srcGlobs = map (pathToTextUnsafe . (</> ("src" </> "**" </> "*.purs")) . dirFor) dependencies
   procs "purs" ("compile" : srcGlobs) empty
 
+data BowerInfoRepo = BowerInfoRepo
+  { url :: Text
+  } deriving (Show, Eq, Generic, Aeson.FromJSON)
+
+data BowerInfo = BowerInfo
+  { bower_name         :: Text
+  , bower_repository   :: BowerInfoRepo
+  , bower_dependencies :: Map.Map Text Text
+  , bower_version      :: Text
+  } deriving (Show, Eq, Generic)
+instance Aeson.FromJSON BowerInfo where
+  parseJSON = Aeson.genericParseJSON Aeson.defaultOptions
+    { fieldLabelModifier = drop 6
+    }
+
+data BowerOutput = BowerOutput
+  { latest :: BowerInfo
+  } deriving (Show, Eq, Generic, Aeson.FromJSON)
+
+addFromBower :: String -> IO ()
+addFromBower name = do
+  let bowerProc = inproc "bower" [ "info", T.pack name, "--json", "-l=error" ] empty
+  result <- fold <$> shellToIOText bowerProc
```

Comment:
Instead of shelling out to `bower`, you could let the user run the `info` command and pipe it in here. We already try to use some of the Bower JSON formats in `purs publish` for example, so I think this approach makes sense here too.

Reply:
But in the end the user wants what happens here right? I wouldn't like to have to look up how to awkwardly use this, and at best, many users would be copy pasting an example command to their terminal and changing it.